### DISCUSSION TITLE
"Data Quality" Scoring of Block

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#17548] [Q&A / Support]
Is it possible to have a "Data Quality" field check instances of a `Block` field? 

For example if I have a `Block` with a `name` and `age` and I want to have the quality score based on whether `name` has been filled in for each instance of the Block.

The "recommended fields check" does not offer these fields. It does however offer nested fields that appear in _Bricks_.
![image](https://github.com/user-attachments/assets/05bfa1ef-912c-4ff8-8d36-daf5f93b11ae)
the Block itself shows up but not the fields within:
![image](https://github.com/user-attachments/assets/46fe4487-075c-497d-bf6a-15d6fa443e8b)

**Comment by fashxp** [comment]
With data quality management you can use the symfony expressions to define some validation rules for the fields within blocks. Depending on the use case that might be a bit clumsy (due to the additional dimensions blocks open up), so you probably also want to consider implementing a [custom rule definition](https://pimcore.com/docs/platform/Data_Quality_Management/Customization/Custom_Rule_Definition) which does exactly what you need for your data model.

**Reply by cancan101** [reply]
Are there any plans to have the "recommended fields check" handle Blocks without needing custom rules / expressions?

For comparison, this is how Bricks are handled now, where the sub fields are displayed, though perhaps that doable there since n=0/1:
![image](https://github.com/user-attachments/assets/df93f159-7b42-405f-8624-77696df27de1)

**Reply by fashxp** [reply]
you gave already the answer yourself. 
with object bricks it is doable, because we have one-to-one. with blocks/flieldcollections we have a one-to-many - which is impossible to define with that kind of definition.
